hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/expiry/ExpirySystemImpl.java

```diff
@@ -171,6 +177,10 @@ public final void add(Data key, long ttl, long maxIdle,
             expiryTime = nextExpirationTime(ttl, maxIdle, now, lastUpdateTime);
         }
 
+        if (disableMaxIdle) {
+            expiryTime = nextExpirationTime(ttl, Long.MAX_VALUE, now, lastUpdateTime);
```

Comment:
what will happen to these expiry-disabled-records when source cluster dies? Did you come to a conclusion?

Reply:
If also TTL isn't configured, they will live forever. Also see the tests in EE.

Reply:
My idea is to disablingMaxIdle/enablingExpiryReplication should be configurable. If user wish to use this, we can suggest also configuring TTL. For example, user can set max idle to 5 min, ttl to 120 min. If this feature is enabled, even if an entry isn't tocuhed, it will at most live 120 mins.